### PR TITLE
Moving Ui message.js hide speed and timeout into variables for easier…

### PR DESCRIPTION
**Pull request by edward-simpson**
… overriding

### Description (*)
Magento Ui message speed and timeout values should be variables instead of hardcoded so they can be overwritten without having to write a whole mixin overwriting the entire `onHiddenChange` function

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Ui/view/frontend/web/js/view/messages.js [hunk 1]
```diff
@@ -68,8 +70,8 @@ define([
             // Hide message block if needed
             if (isHidden) {
                 setTimeout(function () {
-                    $(self.selector).hide('blind', {}, 500);
-                }, 5000);
+                    $(self.selector).hide('blind', {}, self.hideSpeed);
```
Comment (VladimirZaets) [inline app/code/Magento/Ui/view/frontend/web/js/view/messages.js]: Can you please .bind context to timeout callback function and remove `self` usage?

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @edward-simpson. Thanks for collaboration.
Please take a look at my review comments.

File: app/code/Magento/Ui/view/frontend/web/js/view/messages.js (hunk 1 above)
Comment (edward-simpson) [reply app/code/Magento/Ui/view/frontend/web/js/view/messages.js]: Hey Vladimir,

Like this?

```
        onHiddenChange: function (isHidden) {
            // Hide message block if needed
            if (isHidden) {
                setTimeout(function () {
                    $(this.selector).hide('blind', {}, this.hideSpeed);
                }.bind(this), this.hideTimeout);
            }
        }
```

**Review by VladimirZaets — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-6387](https://jira.corp.magento.com/browse/ENGCOM-6387) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

**Review by krzksz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @krzksz, thank you for the review.
[ENGCOM-6387](https://jira.corp.magento.com/browse/ENGCOM-6387) has been created to process this Pull Request

**Pull request merged**